superset/assets/src/dashboard/actions/dashboardLayout.js

```diff
@@ -176,11 +196,8 @@ export function handleComponentDrop(dropResult) {
       dispatch(moveComponent(dropResult));
     }
 
-    const { dashboardLayout: undoableLayout } = getState();
-
     // if we moved a child from a Tab or Row parent and it was the only child, delete the parent.
     if (!isNewComponent) {
-      const { present: layout } = undoableLayout;
```

Comment:
I think this should be an `up-to-date` version of state (i.e., after the possible `dispatch`es above). If you need it above then I would call `getState()` again down here so it's not stale.

Reply:
very important. so i reverted this change, and in the earlier place just get a value from a needed field.